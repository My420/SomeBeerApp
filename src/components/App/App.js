import React from 'react';
import { Route, Switch } from 'react-router-dom';
import HeaderCreator from '../HeaderCreator/HeaderCreator';
import styles from './App.module.scss';
import Container from '../Container/Container';

const App = () => {
  // eslint-disable-next-line no-console
  console.log('render ===== App');
  return (
    <React.Fragment>
      <Container className={styles.container}>
        <HeaderCreator />
      </Container>
      <Switch>
        <Route exact path="/" render={() => <h2>Main</h2>} />
        <Route path="/search" render={() => <h2>Search</h2>} />
        <Route path="/random" render={() => <h2>Random</h2>} />
        <Route render={() => <h2>404 error</h2>} />
      </Switch>
    </React.Fragment>
  );
};
export default App;
