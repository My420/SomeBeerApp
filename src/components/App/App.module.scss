@import '../../sass/variables.scss';

.container {
  background-color: $menu-bar-color;
}
