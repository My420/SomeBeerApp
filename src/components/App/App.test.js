import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('Test <App /> component', () => {
  const wrapper = shallow(<App />);

  test('should correct render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
