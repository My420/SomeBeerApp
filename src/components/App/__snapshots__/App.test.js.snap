// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <App /> component should correct render component 1`] = `
<div
  className="wrapper"
>
  <Container
    className="container"
  >
    <ResizeWrapper(HeaderCreator) />
  </Container>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/catalog"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/favorite"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/cart"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/id/:id"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(ResizeWrapper(RoulettePage))",
          "type": [Function],
        }
      }
      exact={true}
      path="/roulette"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/about"
    />
    <Route
      component={[Function]}
    />
  </Switch>
  <Container
    className="container"
  >
    <Footer />
  </Container>
</div>
`;
